### DISCUSSION TITLE
posts/drop-file-mvvm/

### DISCUSSION THREAD

**Discussion by giscus** [BYJRK/byjrk.github.io#11] [Announcements]
# posts/drop-file-mvvm/

在 WPF 中，实现文件拖入功能并不难，但是想要遵守 MVVM 模式，恐怕需要稍微多花费一点心思。

https://blog.coldwind.top/posts/drop-file-mvvm/

**Comment by ColoCoffee** [comment]
你好, 在我自己尝试的时候发现如果我的程序只有一个mainwindow,其他的页面都是通过导航到不同的UserControl这种方式实现跳转,这个时候如果mainwindow不允许拖放那么其他的页面都无法拖放. 而且哪怕我已经跳转到别的页面优先执行的是mainwindow的拖放事件, 这种问题该怎么解决?

**Reply by ColoCoffee** [reply]
我想要实现的是不同的页面处理他们自己不同的拖放事件, 而不是全部是MainWindow的
